iml-gui/crate/src/components/modal.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{components::font_awesome, generated::css_classes::C};
+use seed::{prelude::*, *};
+
+#[derive(Default)]
+pub struct Model {
+    pub open: bool,
+}
+
+#[derive(Clone)]
+pub enum Msg {
+    KeyDown(u32),
+    Close,
+}
+
+pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
+    match msg {
+        Msg::KeyDown(code) => {
+            if code == 27 {
```

Comment:
Is there a constant we can use to indicate what 27 is?

Reply:
Nope. This should really be part of seed, but I'll add a module for keycodes.

Reply:
Was just curious. I'll +1.